File: ATB_1.1_src/src/main/java/edu/stanford/isis/atb/domain/template/Numerical.java
/*
*  Copyright Northwestern University
*  Copyright Stanford University (ATB 1.0 and ATS 1.0)
*
*  Distributed under the OSI-approved BSD 3-Clause License.
*  See http://ncip.github.com/annotation-and-image-markup/LICENSE.txt for details.
*/



package edu.stanford.isis.atb.domain.template;

import org.hibernate.validator.constraints.NotEmpty;
import org.simpleframework.xml.Attribute;
import org.simpleframework.xml.Root;

/**
 * @author Vitaliy Semeshko
 */

@Root(name="Numerical", strict=false)
public class Numerical extends AbstractRemovableElement {

	@Attribute(required=false)
	private double value;
	
	@NotEmpty(message = "{numerical.ucumString.notEmpty}")
	@Attribute(required=false)
	private String ucumString;
	
	@Attribute(required=false)
	private Operator operator;

	public double getValue() {
		return value;
	}

	public void setValue(double value) {
		this.value = value;
	}

	public String getUcumString() {
		return ucumString;
	}

	public void setUcumString(String ucumString) {
		this.ucumString = ucumString;
	}

	public Operator getOperator() {
		return operator;
	}

	public void setOperator(Operator operator) {
		this.operator = operator;
	}
	
	@Override
	public String toString() {
		return "Numerical: " + value;
	}
	
}

File: AimPlugin3.0.4/AIM.Annotation/Template/Numerical.cs
#region License

//L
// 2007 - 2013 Copyright Northwestern University
//
// Distributed under the OSI-approved BSD 3-Clause License.
// See http://ncip.github.com/annotation-and-image-markup/LICENSE.txt for details.
//L

#endregion

using System.Xml.Serialization;

namespace AIM.Annotation.Template
{
	public class Numerical
	{
		private ComparisonOperator _operator;

		public Numerical()
		{
			OperatorSpecified = false;
		}

		[XmlAttribute("value")]
		public double Value { get; set; }

		[XmlAttribute("ucumString")]
		public string UcumString { get; set; }

		[XmlAttribute("operator", Namespace = TemplateContainer.Namespace)]
		public ComparisonOperator Operator
		{
			get { return _operator; }
			set
			{
				_operator = value;
				OperatorSpecified = true;
			}
		}

		[XmlIgnore]
		public bool OperatorSpecified { get; set; }
	}
}
